tests/Doctrine/Tests/ORM/Tools/Pagination/LimitSubqueryOutputWalkerTest.php

```diff
@@ -274,5 +274,24 @@ public function testLimitSubqueryWithColumnWithSortDirectionInName()
             $query->getSQL()
         );
     }
+
+    public function testLimitSubqueryWithOrderByInnerJoined()
+    {
+        $query = $this->entityManager->createQuery(
+            'SELECT b FROM Doctrine\Tests\ORM\Tools\Pagination\BlogPost b JOIN b.author a ORDER BY a.name ASC'
+        );
+
+        $query->setHint(Query::HINT_CUSTOM_OUTPUT_WALKER, 'Doctrine\ORM\Tools\Pagination\LimitSubqueryOutputWalker');
+
+
+        // current output is:
+        // SELECT DISTINCT id_0, a1_.name FROM (SELECT b0_.id AS id_0, b0_.author_id AS author_id_1, b0_.category_id AS category_id_2 FROM BlogPost b0_ INNER JOIN Author a1_ ON b0_.author_id = a1_.id) dctrn_result ORDER BY a1_.name ASC
+        //
+        // The problem is a1_.name is not output from the subquery, so it cannot be included in the ORDER BY clause.
```

Comment:
a1_.name is not being output in the select list when LimitSubqueryOutputWalker calls $innerSql = parent::walkSelectStatement($AST);